### PR TITLE
Resolves #24 game logo and icon

### LINKED ISSUE #24: Create a logo
**Basic idea/big picture**
An image that identifies this project at first glance. It should be used everywhere the the project name is shown.

**Details**
On a screenshot I saw that the application window uses the default Java icon. That should be replaced with an own application symbol. Such an image can also be used on the GitHub project.

It should be reasoable large (like 256x256) to be able to scale down for different purposes. The image must iconify the idea of the game. For small areas like the application icon it must be readable at low resolutions like 24x24 pixels.

**Technical background**
Maybe a single image format is not enough, but PNG is widely accepted and supports all commons (raster) image features (like transparency). For editing a layered format may be better or a vector graphics like SVG should be used if the content fits.

**Related issues/Prerequisites**
In #12 the README will be filled. This logo should also appear there.

### PR DESCRIPTION
**Pull request by Baret**
PR for #24 to create logo(s) and icon(s) to be used (at least) as
- Application icon
- splash screen/intro (later maybe as intro [animation](https://github.com/Hexworks/zircon/tree/master/zircon.core/src/commonMain/kotlin/org/hexworks/zircon/api/animation))
- Discord server icon

At least the ingame logo needs to be ASCII-art in a format zircon can use ([see here](https://hexworks.org/zircon/docs/2018-11-22-resource-handling)). But to make clear that the game uses that visual style it would be nice to also have all other logo resources be ASCII-art or hinting in that direction.

### CODE REVIEW

**Comment by Baret** [comment]
I created some really basic logos in different aspect ratios. These can sure be extended/improved. But it's at least something.

The bigger problem I see is how to get a nice logo/animation as splash screen ingame. I tried a programmatically drawn one and creating one with REXPaint. Both ways are not so easy...

**Comment by Baret** [comment]
This is just a rough first try on a splash screen and some artwork. The basics are there (like an application icon, different logos that might be used anywhere) and I added the GIMP raw data to be able to edit it well later.

But for now I don't want to waste much more time on "this art stuff". It may be improved later.

File: game/ui/src/main/kotlin/de/gleex/pltcmd/game/ui/composites/GameLogo.kt [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        val bottomRightCorner = Position.create(logoSize.width - 1, logoSize.height - 1).withRelative(logoOffset)
+
+        mainLayer.drawMainLine(topLeftCorner, lowerPoint, upperPoint, bottomRightCorner)
+
+        mainLayer.draw(
+                CharacterTileStrings
+                        .newBuilder()
+                        .withText("p l t")
+                        .build(),
+                Position.create(horizontal - 6, logoSize.height / 2).withRelative(logoOffset))
```
Comment (Loomie) [inline game/ui/src/main/kotlin/de/gleex/pltcmd/game/ui/composites/GameLogo.kt:54]: `logoSize.height / 2` is calculated three times. Create a variable for the middle of the height.

Comment (Loomie) [reply game/ui/src/main/kotlin/de/gleex/pltcmd/game/ui/composites/GameLogo.kt:54]: Or create an extension function `Int.half()`

File: game/ui/src/main/kotlin/de/gleex/pltcmd/game/ui/composites/GameLogo.kt [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+/**
+ * The logo used as splash screen. Intended to be drawn onto the full screen, but might also be used in other places
+ */
+object GameLogo {
+    private val style = StyleSet.create(UiOptions.THEME.primaryForegroundColor, ANSITileColor.BLACK)
+    private val styleFadingShadow = style
+            .withModifiers(Modifiers.fadeIn(20, 2000, true))
+            .withForegroundColor(UiOptions.THEME.primaryBackgroundColor)
+            .withBackgroundColor(ANSITileColor.BLACK)
```
Comment (Loomie) [inline game/ui/src/main/kotlin/de/gleex/pltcmd/game/ui/composites/GameLogo.kt:26]: A black shadow on a black background is not that noticable ;) Using the `primaryBackgroundColor` as forground color is intended?!

File: pom.xml [hunk 3]
```diff
@@ -32,7 +32,7 @@
 		<kotlin.compiler.jvmTarget>1.8</kotlin.compiler.jvmTarget>
 		<kotlin.version>1.3.72</kotlin.version>
 
-		<zircon.version>2020.1.3-PREVIEW</zircon.version>
+		<zircon.version>2020.1.5-PREVIEW</zircon.version>
 		<!-- Should match the version used by Zircon -->
 		<cobalt.version>2020.0.15-PREVIEW</cobalt.version>
```
Comment (Loomie) [inline pom.xml:37]: > Should match the version used by Zircon

Comment (Loomie) [reply pom.xml]: fixed

File: game/ui/src/main/kotlin/de/gleex/pltcmd/game/ui/composites/GameLogo.kt (hunk 1 above)
Comment (Loomie) [reply game/ui/src/main/kotlin/de/gleex/pltcmd/game/ui/composites/GameLogo.kt:54]: extracted variable

File: game/ui/src/main/kotlin/de/gleex/pltcmd/game/ui/composites/GameLogo.kt (hunk 2 above)
Comment (Baret) [reply game/ui/src/main/kotlin/de/gleex/pltcmd/game/ui/composites/GameLogo.kt:26]: Yes it is intended.

File: game/options/src/main/kotlin/de/gleex/pltcmd/game/options/UiOptions.kt [hunk 4]
```diff
@@ -13,6 +13,7 @@ object UiOptions {
                     withSize(WINDOW_WIDTH, WINDOW_HEIGHT).
                     withTitle("PltCmd").
                     withDefaultTileset(DEFAULT_TILESET).
+                    withIcon("icon/icon.png").
```
[the remark this answers is not in the document]
Comment (Baret) [reply game/options/src/main/kotlin/de/gleex/pltcmd/game/options/UiOptions.kt:16]: actually I did not notice :D

**Comment by Baret** [comment]
The aspect ratio is 1.6 so it should be the same as the icon. maybe with a border it would look more alike.
I will give it a quick try...

The colors are not quite as intended but I think we ca adjust the details later.

**Comment by Baret** [comment]
Itried putting a border around the splash screen logo. but it looks awkward...

Have another review :)

**Pull request merged**